oauth2client/client.py

```diff
@@ -1784,7 +1784,9 @@ class OAuth2WebServerFlow(Flow):
   """
 
   @util.positional(4)
-  def __init__(self, client_id, client_secret, scope,
+  def __init__(self, client_id,
+               client_secret=None,
+               scope=None,
```

Comment:
Maybe I'm misunderstanding, but it seems like this decorator is supposed to declare the *maximum* number of positional arguments. From the docstring:

> A decorator to declare that only the first N arguments my be positional.

I deliberately want to make sure that this changeset doesn't break any existing users of this library, so I think we still want to support passing up to 4 positional arguments.